### ISSUE TITLE
Explore how to inject Logger interface on Angular

### ISSUE THREAD

**Issue by doup** [mobilejazz/harmony-typescript#21]
We need to add an `InjectionToken` or something similar to be able to use the `Logger` interface on Angular projects. For now the components can ask for the `AbstractLogger` and that will do the trick. E.g.:

```typescript
// In the provider
{
    provide: AbstractLogger,
    useFactory: () => new DeviceConsoleLogger(),
}

// In the component
constructor(
    protected logger: AbstractLogger,
) {}
```

This would be something to add into the Angular package.

**Comment by doup** [comment]
It's not that complicated @doup… why not convert the `Logger` into an abstract class???

**Comment by joselufo** [comment]
Another issue for the same thing #87 

Let's resolved in #87 and then we close all the issues.

**Comment by orioljp** [comment]
This issue has been moved to Asana.
https://app.asana.com/0/1109863238977521/1203172481263555/f
Closing it.
